source/exe/main_common.cc

```diff
@@ -75,6 +75,8 @@ MainCommonBase::MainCommonBase(OptionsImpl& options, Event::TimeSystem& time_sys
     server_ = std::make_unique<Server::InstanceImpl>(
         options_, time_system, local_address, test_hooks, *restarter_, *stats_store_,
         access_log_lock, component_factory, std::move(random_generator), *tls_);
+
+    server_->RegisterMutexContentionStats();
```

Comment:
Agreed, this should be flag-enabled, since I could imagine scenarios where users of Envoy don't want to incur the (slight) mutex contention overhead. Will do.